### PR TITLE
Update AtkUnitBase

### PR DESCRIPTION
**Pull request by Haselnussbomber**
Going through AtkUnitBase again, adding flag fields as they are used by the game and doing some cleanup.

Along with the flags, a couple notable changes:
- Added `AtkResNode.IsVisible` function
  It looks like someone found a function to check if an AtkResNode is visible. It does a little bit more than just checking the flag, so I included it. This obviously changes it from being a property to being a function.
- Added `AtkUnitBase.IsFullyLoaded` (vf65) and removed `AtkUldManager.SearchNodeById` wrapper
  This virtual function checks if all resources (uld and optional scd) are loaded and if the UldManager loaded the uld and all the textures it needs. This function should be the preferred way of checking if an addon was fully loaded going forward.
- Added `AtkExternalInterface` vfuncs 1 (CallHandler) and 2 (PlaySoundEffect).

### CODE REVIEW

**Comment by Haselnussbomber** [comment]
Should be enough now. I can't really figure out all possible bitflags or give those fields proper names.

Buuut, I've successfully made a custom addon without loading an uld file. (Nodes added with the help of KamiToolKit.)
Code: https://gist.github.com/Haselnussbomber/7a9c1f3b60ab86ea9d97d4a44b40e4a7

Screenshot

File: FFXIVClientStructs/FFXIV/Component/GUI/AtkUnitBase.cs [hunk 1]
```diff
@@ -21,16 +21,86 @@ public unsafe partial struct AtkUnitBase : ICreatable {
...
+    /// The following scds can be loaded:
+    /// <code>
+    /// 0 = sound/system/SE_UI.scd
+    /// 1 = sound/system/SE_GS.scd
+    /// 2 = sound/system/SE_TTriad.scd
+    /// 3 = sound/system/SE_EMJ.scd
+    /// 4 = sound/system/SE_10thMG.scd
+    /// </code>
+    /// </summary>
+    [FieldOffset(0x1B4)] public byte ScdResourceIndex;
```
Comment (wolfcomp) [inline FFXIVClientStructs/FFXIV/Component/GUI/AtkUnitBase.cs:98]: The remark on this doesn't fully make sense unless SE_UI is part of the not fully loaded base

**Review by wolfcomp — COMMENTED** [review]
Just one thing I noticed going through the functions for this

File: FFXIVClientStructs/FFXIV/Component/GUI/AtkUnitBase.cs (hunk 1 above)
Comment (Haselnussbomber) [reply FFXIVClientStructs/FFXIV/Component/GUI/AtkUnitBase.cs:97]: Removed it.

**Pull request merged**